impl/src/main/java/org/jboss/weld/bean/attributes/BeanAttributesFactory.java

```diff
@@ -171,7 +171,7 @@ protected void initQualifiers(Set<Annotation> qualifiers) {
             if (qualifiers.isEmpty()) {
                 this.qualifiers = DEFAULT_QUALIFIERS;
             } else {
-                ArraySet<Annotation> normalizedQualifiers = new ArraySet<Annotation>(qualifiers.size() + 2);
+                Set<Annotation> normalizedQualifiers = new HashSet<Annotation>(qualifiers.size() + 2);
```

Comment:
The point of this change is that `normalizedQualifiers` are send into `SharedObjectCache` as an immutable set?
If so, we need to do it differently (`ImmutableSet.copyOf`), because `Builder` cannot do `remove()` method (L182)